### DISCUSSION TITLE
Custom InterpolatedStringHandler with string.Concat like overloads

### DISCUSSION THREAD

**Discussion by ladeak** [dotnet/csharplang#8306] [Q&A] [answered]
The compiler seems to choose a `string.Concat` overload to create strings with only a few parts (instead of using the DefaultInterpolatedStringHandler), as it seems to be more performant for some scenarios.

When I create a custom interpolated string handler, is there a way to tell the compiler to use a certain method (and its overloads) for similar scenarios?

**Comment by stephentoub** [comment]
> When I create a custom interpolated string handler, is there a way to tell the compiler to use a certain method (and its overloads) for similar scenarios?

Can you provide an example?

**Reply by ladeak** [reply]
I create an InterpolatedStringHandler that assembles that appends URL path segments together while also taking care of using the starting or ending `/`-s on each segment (trimming or appending them as needed).

```$"{BaseUrl}/path/entity/{Id}/type/{TypeValue}"```

But I notice that when I only have a few of these segments, the performance is less attractive. I wonder how the DefaultInterpolatedStringHandler manages this situation but from the IL it looks the compiler invokes `string.Concat` instead.

Hence I wonder if I create a method that has a signature like `string.Concat`, but has my custom implementation, then is there a way to tell the compiler to use this method instead of the my InterpolatedStringHandler?

**Reply by stephentoub** [reply]
@333fred, does OverloadResolutionPriority handle this?

**Reply by 333fred** [reply, marked as answer]
No, `OverloadResolutionPriority` wouldn't help with something like this. Using an interpolated string as a string is a special case, and handled specially by the compiler because it can guarantee observed semantics.

**Reply by ladeak** [reply]
So as I understand there is no direct way to achieve this. And my suspect is that some performance is lost by the `ArrayPool<T>` renting and return. I would do stackalloc normally to avoid this with a fixed buffer, but similar to `DefaultInterpolatedStringHandler` cannot really do that in a `ref struct`.

EDIT: I think that I could replace the stackalloc with an inline array though - that seems to be a workable solution.
